torch/onnx/_internal/diagnostics/rules.yaml

```
@@ -354,6 +354,42 @@
     deprecated: false
     tags: []
 
+- id: FXE0014
+  name: find-opschema-matched-symbolic-function
+  short_description:
+    text: Find the OnnxFunction that matches the input dtypes by comparing them with their opschemas.
+  full_description:
+    text: "Find the OnnxFunction that matches the input dtypes by comparing them with their opschemas.
+      A warning will be issued if the matched OnnxFunction is not an exact match."
+    markdown: |
+      When an ATen/Custom operator is registered and needs to be dispatched to an OnnxFunction, the input
+      dtypes of the ATen/Custom operator are compared with the input dtypes of the OnnxFunction opschemas
+      to find a match. However, if a perfect/exact match is not found, the dispatcher will attempt to find
+      the nearest match with the highest number of input dtypes matching the OnnxFunction opschemas, while
+      issuing a warning.
+
+      There are two types of level that can be triggered in this rule:
+
+      1. NOTE: A perfect match is found, and no warning is issued.
```

```suggestion
      1. NONE: A perfect match is found, and no warning is issued.
```
Should be `NONE`. I hate sarif level that NOTE actually also means problem, just not as severe as warning.

---

Actually don't worry about it here. I found out Sarif extension render it as `Level: note, Kind informational`, although the json file lists it as `Level: none, Kind informational`. We should just drop NONE level from our diagnostics API. And point NOTE to sarif None ...... 💀

---

So keep it for now, and there will be a refactor for NOTE/NONE?

---

yep 🚢